test/libsolidity/SMTChecker.cpp

```diff
@@ -116,13 +116,15 @@ BOOST_AUTO_TEST_CASE(division)
 				if (a == 0) {
 					return 0;
 				}
+				// TODO remove when SMTChecker sees that this code is the `else` of the `return`.
+				require(a != 0);
```

Comment:
Does the `require` prevent Z3 from trying to solve with `a == 0`?

Reply:
Yes. This should be the case because `a == 0` returns, but the SMTChecker does not see that `a != 0` should be true here (as an `else` of the `if` because of `return`), that's why I added it explicitly.